--- canton/community/ledger-api/src/main/protobuf/com/daml/ledger/api/v2/interactive/interactive_submission_service.proto	2025-08-19 23:17:45.942120404 +0300
+++ canton/community/ledger-api/src/main/protobuf/com/daml/ledger/api/v2/interactive/interactive_submission_service_p.proto	2025-08-19 23:53:17.040538973 +0300
@@ -140,19 +140,19 @@
 message PrepareSubmissionResponse {
   // The interpreted transaction, it represents the ledger changes necessary to execute the commands specified in the request.
   // Clients MUST display the content of the transaction to the user for them to validate before signing the hash if the preparing participant is not trusted.
-  PreparedTransaction prepared_transaction = 1;
+  // PreparedTransaction prepared_transaction = 1;
   // Hash of the transaction, this is what needs to be signed by the party to authorize the transaction.
   // Only provided for convenience, clients MUST recompute the hash from the raw transaction if the preparing participant is not trusted.
   // May be removed in future versions
-  bytes prepared_transaction_hash = 2;
+  bytes prepared_transaction_hash = 1;
 
   // The hashing scheme version used when building the hash
-  HashingSchemeVersion hashing_scheme_version = 3;
+  HashingSchemeVersion hashing_scheme_version = 2;
 
   // Optional additional details on how the transaction was encoded and hashed. Only set if verbose_hashing = true in the request
   // Note that there are no guarantees on the stability of the format or content of this field.
   // Its content should NOT be parsed and should only be used for troubleshooting purposes.
-  optional string hashing_details = 4;
+  optional string hashing_details = 3;
 }
 
 message Signature {
@@ -325,7 +325,7 @@
   uint32 mediator_group = 4;
   string transaction_uuid = 5;
   uint64 preparation_time = 6;
-  repeated InputContract input_contracts = 7;
+  int32 input_contracts_count = 7;
 
   /*
    * Where ledger time constraints are imposed during the execution of the contract they will be populated
@@ -376,10 +376,10 @@
   string version = 1;
   // Root nodes of the transaction
   repeated string roots = 2;
-  // List of nodes in the transaction
-  repeated Node nodes = 3;
   // Node seeds are values associated with certain nodes used for generating cryptographic salts
-  repeated NodeSeed node_seeds = 4;
+  repeated NodeSeed node_seeds = 3;
+  // Number of nodes in the transaction
+  int32 node_count = 4;
 }
 
 message GetPreferredPackageVersionRequest {
